### PR TITLE
CLDR-15368 Enable API to replace WHAT_GETROW

### PR DESCRIPTION
**Pull request by btangmu**
-Enable Survey Tool to work correctly with TABLE_USES_NEW_API = true

-Keep TABLE_USES_NEW_API = false for now pending further testing

-Make VoteAPI produce json more like DataSection.getRow

-Supply numerous data items that were missing from VoteAPI json

-Remove duplicate VoteResolver in VoteAPI

-Distinguish front-end votingResults from back-end voteResolver

-Lazy initialization of PathHeader alphabetic collator to avoid CLDRConfig conflict

-A few front end changes, no need for displaySets.default; fix getSingleRowUrl; simplify testItem.cause

-Comments

CLDR-15368

- [ ] This PR completes the ticket.

### CODE REVIEW

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java [hunk 1]
```diff
@@ -2116,11 +2107,24 @@ public static String getMetazonePageTerritory(String source) {
...
 
+    private static synchronized void initializeAlphabetic() {
+        // Lazy initialization: don't call CLDRConfig.getInstance() too early or we'll get
+        // "CLDRConfig.getInstance() was called prior to SurveyTool setup" when called from
+        // com.ibm.ws.microprofile.openapi.impl.core.jackson.ModelResolver._addEnumProps
+        if (alphabetic == null) {
+            alphabetic = CLDRConfig.getInstance().getCollatorRoot();
+        }
+    }
+
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java:2127]: make it a [Bill Pugh singleton](https://en.wikipedia.org/wiki/Singleton_pattern#Lazy_initialization) instead?

**Review by srl295 — APPROVED** [review]
+1 ! Lots of good work here, data wrangling and paying down tech debt especially 🎉

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java (hunk 1 above)
Comment (btangmu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java:2127]: you mean like
```
public final class SingletonBillPugh {
	private static class SingletonHelper {
		private static final SingletonBillPugh INSTANCE = new SingletonBillPugh();
	}

	private SingletonBillPugh() {
	}

	public static SingletonBillPugh getInstance() {
		return SingletonHelper.INSTANCE;
	}
}
```
?

Comment (btangmu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java:2127]: Let's discuss -- in the meantime I'll merge the PR as-is -- thanks for the review!

**Pull request merged**